Documentation/contributors/design-proposals/installed-api.md

```diff
@@ -0,0 +1,282 @@
+# Installed Operator API
+
+Status: Pending
+
+Version: Alpha
+
+Implementation Owner: TBD
+
+## Motivation
+
+Today, information about operators configured to watch a given namespace is surfaced via "copied" `ClusterServiceVersion`s (`CSV`s). This means that for each __<CSV, watched namespace>__ pair, OLM must manage a copied CSV instance. Additionally, `CSV`s often contain large amounts of metadata, most notably a [base64 encoded icon](https://github.com/operator-framework/operator-lifecycle-manager/blob/18b6c0d2edba2534b4726138b225fff066aae99a/manifests/0000_50_olm_03-clusterserviceversion.crd.yaml#L114). As of OpenShift 4.2, there are more than a dozen namespaces that globally installed operators have their `CSV`s copied to, which causes OLM's `CSV` cache (and memory consumption) to grow large (~150MB) with a fairly low number of operators installed (~5). 
+
+Due to these factors, an additional means of providing this information is required.
+
+## Proposal
+
+Define a new read-only API resource, `installed.packages.operators.coreos.com`, that surfaces metadata about an operator installed and configured to watch a namespace. Back that API by an extension api-server that _synthesizes_ those resources on request.
+
+### Naming
+
+The resource will share a name with the `CSV` of the installed operator it's tracking.
+
+### Namespaces
+
+The resource will be namespaced. Additionaly, it will exist only when an operator that provides a package matching its `metadata.name` is installed and configured to watch its namespace.
+
+### Spec and Status
+
+The resource __will not__ define a `spec` field because it's intended to aggregate the state of existing resources. In other words, there is no meaning in declaring a _desired state_. However, It __will__ define a `status` field that embeds the latest `CSV` and `Subscription`. Embedding entire resources instead of their spec or status directly allows tracked resources to be added or removed more easily in subsequent API changes.
```

Comment:
You’ve read over the kube api conventions more recently than I - isn’t it fine to omit spec and status in cases like this? Should we just not have them?

Reply:
Here, I felt that `status` would imply intent; that it would be easier for users to understand that they can't modify the listed resources. After re-reading the api conventions' [section on subresources](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#types-kinds), I think omitting the `status` and pulling the tracked resources to the top-level (as I think you are alluding to) is fine as well.

Do you have a preference?

Reply:
I think omitting sounds good but it might be one of those things where you try it and see how it looks?